functionalDB/chapter.md

```diff
@@ -850,21 +835,18 @@ Having the query plan to be a function that accepts a database as an argument is
 ````
 #### Phase 3 - Execution of the plan
 
-In general, the execution plan of a query in our database is:
-
+We saw in the previous phase that the query plan we construct ends by calling *single-index-query-plan*. This function will:
 
 1. Apply each predicate clause on an index (each predicate on its appropriate index level)
-2. Perform an AND operation between the above results 
-3. Return a structure that simplifies the reporting of the results
-
-The function *single-index-query-plan* implements this description, where the first two steps are done in *query-index*, and the last is handled in *bind-variables-to-query*
+2. Perform an AND operation across the results 
+3. Merge the results into a simpler data structure
 
 ````clojure
 (defn single-index-query-plan [query indx db]
    (let [q-res (query-index (ind-at db indx) query)]
      (bind-variables-to-query q-res (ind-at db indx))))
 ````
-Let’s go deeper into the rabbit's hole and take a look at the *query-index* function, where a query and data give birth to results. 
+Let’s go deeper into the rabbit's hole and take a look at the *query-index* function, where our query finally begins to yield some data:

```

Comment:
I will change it to yield some results and not yield some data, all along the way we are doing data transformations